src/app/shared/session-storage/session-storage.service.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Rx';
+
+@Injectable()
+export class SessionStorageService {
+
+  private token: string;
+  constructor() {}
+
+  public getToken(key: string) {
```

Comment:
That's awesome but the name of the function should change as well. Instead of `getToken` it should be `getValue`